File: python/euler3.py
# Project Euler Problem 3

def biggest_factor(x):
    pf = []
    
    while x % 2 == 0:
        pf.append(x)
        x /= 2

    y = 3

    while y*y <= x:
        if x % y == 0:
            pf.append(y)
            x /= y

        else:
            y += 2

    if x != 1:
        pf.append(x)
        
    return max(pf)

print(f'Ans = {int(biggest_factor(600851475143))}')

File: julia/euler3.jl
# Project Euler Problem 3

function factor_list(num)
    f_l = []

    while num%2==0
        push!(f_l,2)
        num/=2
    end

    d = 3

    while d^2 <= num
        if num%d==0
            push!(f_l,d)
            num /= d
        else
            d += 2
        end
    end

    if num != 1
        push!(f_l, num)
    end

    return f_l
end

println("Ans = ", factor_list(600851475143)[end])
